File: src/yanzm/example/adaptersamples/SimpleAdapterSample.java
package yanzm.example.adaptersamples;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

import android.app.ListActivity;
import android.os.Bundle;
import android.widget.SimpleAdapter;

public class SimpleAdapterSample extends ListActivity {
	
    @SuppressWarnings({ "unchecked", "serial" })
	public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        final List<HashMap<String, String>> data = 
            Arrays.asList(
                new HashMap<String, String>() {{
                    put("Title", "我輩は猫である");
                    put("Author", "夏目漱石");
                }},
                new HashMap<String, String>() {{
                    put("Title", "走れメロス");
                    put("Author", "太宰治");
                }}
            );

        String[] from = {"Title", "Author"};
        
        int[] to = {R.id.textview1, R.id.textview2};
        
        SimpleAdapter simpleAdapter = new SimpleAdapter(this, data, R.layout.list_item, from, to);
        setListAdapter(simpleAdapter);
    }
}

File: app/src/main/java/yanzm/example/adaptersamples/SimpleAdapterSample.kt
package yanzm.example.adaptersamples

import android.os.Bundle
import android.widget.ListView
import android.widget.SimpleAdapter
import androidx.appcompat.app.AppCompatActivity

class SimpleAdapterSample : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val listView = ListView(this)
        setContentView(listView)

        val data = listOf(
            mapOf(
                "Title" to "我輩は猫である",
                "Author" to "夏目漱石"
            ),
            mapOf(
                "Title" to "走れメロス",
                "Author" to "太宰治"
            ),
        )
        val from = arrayOf("Title", "Author")
        val to = intArrayOf(R.id.textview1, R.id.textview2)

        listView.adapter = SimpleAdapter(this, data, R.layout.list_item, from, to)
    }
}
